### PR TITLE
Upgrade provider version to 3.7 #147

### PR DESCRIPTION
**Pull request by pimielowski**
## Description

Since there was significant update of AzureRM provider we need to follow up with our stack. 

## Motivation and Context

It was part of issue #147

## How Has This Been Tested?

I tested with all example couple off times with different config
- bootstrap 			
- panorama 			
- transit_vnet_common 	
- transit_vnet_dedicated 
- vmseries 			
- vmseries_scaleset 		
- vnet 				

## Types of changes

- Breaking change (fix or feature that would cause existing functionality to change)

Changing default version of AzureRM provider.

## Checklist

- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes if appropriate.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by rokeno** [comment]
Do you guys need any help with approving this PR?

File: examples/transit_vnet_common/main.tf [hunk 1]
```diff
@@ -79,12 +80,13 @@ module "outbound_lb" {
...
   resource_group_name = azurerm_resource_group.this.name
   enable_zones        = var.enable_zones
+  avzones             = var.avzones
   frontend_ips = {
     outbound = {
       subnet_id                     = lookup(module.vnet.subnet_ids, "subnet-private", null)
       private_ip_address_allocation = "Static"
       private_ip_address            = var.olb_private_ip
-      availability_zone             = var.enable_zones ? null : "No-Zone" # For the regions without AZ support.
+      zones                         = var.enable_zones ? null : var.avzones # For the regions without AZ support.
```
[the remark this answers is not in the document]
Comment (pimielowski) [reply examples/transit_vnet_common/main.tf:89]: You are right, thanks :)

File: modules/loadbalancer/variables.tf [hunk 2]
```diff
@@ -134,3 +134,13 @@ variable "tags" {
   default     = {}
   type        = map(string)
 }
+
```
[the remark this answers is not in the document]
Comment (pimielowski) [reply modules/loadbalancer/variables.tf:137]: Sure, you are right, that one is clearly missing there, thanks!

File: examples/panorama/variables.tf [hunk 3]
```diff
@@ -104,3 +99,107 @@ variable "enable_zones" {
...
+      rules = {}
+    }
+  }
+  ```
+  EOF
+}
+
+variable "allow_inbound_mgmt_ips" {
+  description = <<-EOF
+    List of IP CIDR ranges (like `["23.23.23.23"]`) that are allowed to access management interfaces of VM-Series.
```
[the remark this answers is not in the document]
Comment (pimielowski) [reply examples/panorama/variables.tf]: Fixed

File: examples/panorama/variables.tf [hunk 4]
```diff
@@ -104,3 +99,107 @@ variable "enable_zones" {
...
+    If you use Panorama, include its address in the list (as well as the secondary Panorama's).
```
[the remark this answers is not in the document]
Comment (pimielowski) [reply examples/panorama/variables.tf]: Fixed

**Review by migara — APPROVED** [review]
LGTM 🚀

**Pull request merged**